### PR TITLE
feat: add wrapper to expose stdout, stderr, and exitcode

### PR DESCRIPTION
**Pull request by dreinhardt-terminus**
Installing a wrapper script to wrap subsequent calls of the terraform binary and expose its `STDOUT`, `STDERR`, and `exit code` as outputs named `stdout`, `stderr`, and `exitcode` respectively. This is turned off by default.

### CODE REVIEW

File: src/setup-tflint.js [hunk 1]
```diff
@@ -71,9 +74,44 @@ async function downloadCLI(url) {
   return pathToCLI;
 }
 
+async function installWrapper(pathToCLI) {
+  let source;
+  let target;
+
+  const files = fs.readdirSync(pathToCLI);
+
+  core.debug(files.toString());
```
Comment (bendrucker) [inline src/setup-tflint.js:83]: `files` is never used except in these debug logs.

File: src/setup-tflint.js [hunk 2]
```diff
@@ -83,6 +121,11 @@ async function run() {
 
     const pathToCLI = await downloadCLI(url);
 
+    // Install our wrapper
```
Comment (bendrucker) [inline src/setup-tflint.js:124]:
```suggestion
```

File: wrapper/tflint.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+const { exec } = require('@actions/exec');
+const io = require('@actions/io');
+
+const OutputListener = require('./lib/output-listener');
+const pathToCLI = require('./lib/tflint-bin');
+
+async function checkTflint() {
+  // Setting check to `true` will cause `which` to throw if tflint isn't found
+  const check = true;
+  return io.which(pathToCLI, check);
```
Comment (bendrucker) [inline wrapper/tflint.js:13]:
```suggestion
  // throws if `which` does not find a result
  return io.which(pathToCLI, true);
```

File: wrapper/tflint.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  core.setOutput('exitcode', exitCode.toString(10));
+
+  if (exitCode === 0 || exitCode === 2) {
+    // A exitCode of 0 is considered a success
+    // An exitCode of 2 denotes no errors occurred, but issues found
+    return;
+  }
+
+  // A non-zero exitCode is considered an error
+  core.setFailed(`TFlint exited with code ${exitCode}.`);
```
Comment (bendrucker) [inline wrapper/tflint.js:52]:
```suggestion
  core.setFailed(`TFLint exited with code ${exitCode}.`);
```

**Review by bendrucker — CHANGES_REQUESTED** [review]
Thanks! Couple small comments but I recognize that this is mostly borrowed from `hashicorp/setup-terraform` and kept the requests minimal.

File: dist/index1.js
[the remark this answers is not in the document]
Comment (dreinhardt-terminus) [reply dist/index1.js:1]: I'm just copying setup-terraform for this. Not sure if there is a better/cleaner way.

[the remark this answers is not in the document]
Comment (bendrucker) [reply dist/index1.js:1]: I don't see it being invoked anywhere though, am I missing something?

**Pull request closed without merging**